src/Services/Dominion/GovernmentService.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace OpenDominion\Services\Dominion;
+
+use Carbon\Carbon;
+use OpenDominion\Models\Dominion;
+use OpenDominion\Models\Realm;
+use OpenDominion\Services\Dominion\HistoryService;
+
+class GovernmentService
+{
+    /**
+     * Gets votes for Realm monarchy by Dominion.
+     *
+     * @param Realm $realm
+     * @return array
+     */
+    public function getMonarchVotes(Realm $realm): array
+    {
+        $votes = $realm->dominions->groupBy('monarch_dominion_id');
+
+        $results = [];
+        foreach ($votes as $monarch => $dominions) {
+            if ($monarch != null) {
+                $results[$monarch] = count($dominions);
+            }
+        }
+
+        return $results;
+    }
+
+    /**
+     * Check if a new monarch has been elected for a Realm.
+     *
+     * @param Realm $realm
+     * @return bool
+     */
+    public function checkMonarchVotes(Realm $realm): bool
+    {
+        $currentMonarchId = $realm->monarch->id;
```

Comment:
This line throws:

 ErrorException (E_NOTICE)
Trying to get property 'id' of non-object

When I vote for someone

Reply:
Ahh yeah I changed it due to a bug discovered in ODA, but didn't test it again with null.